src/main/java/com/exasol/matcher/UpcastOnlyCellMatcher.java

```diff
@@ -0,0 +1,116 @@
+        this.explanation = "";
+        if (actual instanceof Number && this.expected instanceof Number) {
+            return checkNumbersAreOnlyUpcasted(actual, this.expected);
+        } else {
+            return true;
+        }
+    }
+
+    @Override
+    public void describeTo(final Description description) {
+        description.appendText("type safely castable to ").appendText(this.expected.getClass().getName());
+    }
+
+    @Override
+    protected void describeMismatchSafely(final T item, final Description mismatchDescription) {
+        mismatchDescription.appendText(this.explanation);
+    }
+
+    private boolean checkNumbersAreOnlyUpcasted(final Object actual, final Object expected) {
+        if (DECIMAL_TYPES.contains(actual.getClass()) && DECIMAL_TYPES.contains(expected.getClass())) {
+            return checkDecimalUpcast(actual);
+        } else if (FLOATING_POINT_TYPES.contains(actual.getClass())
+                && FLOATING_POINT_TYPES.contains(expected.getClass())) {
+            return checkFloatingPointUpcast(actual, expected);
+        } else if (DECIMAL_TYPES.contains(actual.getClass()) && FLOATING_POINT_TYPES.contains(expected.getClass())) {
+            return checkDecimalToFloatUpcast(actual);
+        } else {
+            this.explanation = "\nCan not cast from actual floating point to expected non-floating point type.";
+            return false;
+        }
+    }
+
+    private boolean checkDecimalToFloatUpcast(final Object actual) {
+        final int actualIndex = DECIMAL_TYPES.indexOf(actual.getClass());
+        if (actualIndex <= DECIMAL_TYPES.indexOf(Short.class)) {
+            return true; // we can safely cast a SHORT to a float or double
+        } else if (actualIndex <= DECIMAL_TYPES.indexOf(Integer.class)
+                && this.expected.getClass().equals(Double.class)) {
+            setFloatingPointExplanation();
+            return true; // we can safely cast an Integer to a double
+        } else {
+            setFloatingPointExplanation();
+            return false;
+        }
+    }
+
+    private boolean checkFloatingPointUpcast(final Object actual, final Object expected) {
+        if (FLOATING_POINT_TYPES.indexOf(actual.getClass()) <= FLOATING_POINT_TYPES.indexOf(expected.getClass())) {
+            return true;
+        } else {
+            setActualTypeBiggerThatExpectedExplanation();
+            return false;
+        }
+    }
+
+    private boolean checkDecimalUpcast(final Object actual) {
+        if (DECIMAL_TYPES.indexOf(actual.getClass()) <= DECIMAL_TYPES.indexOf(this.expected.getClass())) {
+            return true;
+        } else {
+            setActualTypeBiggerThatExpectedExplanation();
+            return false;
+        }
+    }
+
+    private void setActualTypeBiggerThatExpectedExplanation() {
+        this.explanation = "\nThe actual type is bigger than the expected. You can disable this check by using the NO_TYPE_CHECK fuzzy-mode.";
```

Comment:
```suggestion
        this.explanation = "\nThe actual type is bigger than the expected. You can disable this check by using the NO_TYPE_CHECK fuzzy-mode.";
```

While I like mitigations in error messages, this is an assertion message and should only contain the relevant results.